iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -363,7 +361,7 @@ cxx::expected<int32_t, IpcChannelError> UnixDomainSocket::createSocket(const Ipc
         return createErrorFromErrnum(socketCall.getErrNum());
     }
 
-    int32_t sockfd = socketCall.getReturnValue();
+    m_sockfd = socketCall.getReturnValue();
```

Comment:
this is a source for future bugs. We might end up with a dangling file descriptor which is not set to `INVALID_FD`.

Reply:
@elBoberido I completely understand you but if you store it in a temporary variable and then use this variable to close the file descriptor again via `closeFd(tempSockFd)` we have there a bigger source for future bugs since `closeFd` needs to invalidate `m_sockfd` and has to set the whole object to `m_isInitialized = false`.

Then you would have to verify in `closeFd` if it is an arbitrary file descriptor or the socket file descriptor and you have to invalidate everything! 

In my opinion this is the cleanest approach and with this abstraction and sufficient tests we should be good to go.

Additionally, in the previous approach you must always assign the sockfd manually after a successful call and with the logic of a source for future bug you could argue that a future developer maybe forgets to assign this and then you would end up with an invalidated file descriptor.

Reply:
I didn't say the previous approach was better, just that this might be improved if thinking a bit more about it

Reply:
How about the following:
- keep the old name and return the file descriptor
- rename the current `closeFileDescriptor` to `closeAndUnlink`
- create a static `closeFileDescriptor` method which takes a file descriptor as parameter
- call that one in this method and in `closeAndUnlink`

With this, it should not be possible to end up with a dangling file descriptor.

Reply:
For me it doesn't make sense to use a creational pattern aka `createSocket()` here. What you describe perfectly fit for something like a factory, where I say "Hey please bake that object for me". In our case it is different. `sockfd` is a member of the object, hence an integral part and nothing that gets created and is then attached to the object. Think of a car that already has tires and which get pumped up aka initalised at a certain point in the lifetime of the car. What do you think fits better `initializeTires()" or `createTires()`? With the current approach it should also not be possible to end up with dangling pointers.

Reply:
Well it's more like a `check if the tires are flat before mounting it on the car` analogy :)

Reply:
oh, and it's possible run twice through `closeFileDescriptor` if the first call fails. That wouldn't be the case if `m_sockfd` is not set in the first place.